src/app/plugins/platform/Docker/plugins/cms/Drupal8/installDrupal.ts

```diff
@@ -36,82 +51,61 @@ export interface InstallDrupalOptions {
   projectType: Project;
 }
 
-/** Installs Drupal from a given project installation type. */
-async function installDrupal({
-  serviceDirectory,
-  documentRoot,
-  projectType,
-}: InstallDrupalOptions): Promise<void> {
-  const needsRename = documentRoot !== 'web';
-
-  // Crash early if the user asked for a non-'web' root for a Pantheon project.
-  // This will help prevent a lot of headaches due to misalignment with the platform's
-  // requirements.
-  if (projectType === pantheonProject && needsRename) {
-    throw new Error(
-      `Pantheon projects do not support '${documentRoot}' as the document root.`,
-    );
-  }
+/**
+ * Installs Drupal from a given project installation type.
+ *
+ * @param {InstallDrupalOptions}
+ * @returns {Promise<void>}
+ */
+async function installDrupal(
+  installOptions: InstallDrupalOptions,
+): Promise<void> {
+  // @todo Revise this to use better syntax options.
+  const projectType = installOptions.projectType;
 
-  await makeDir(serviceDirectory);
+  // Confirm a valid web root was selected for the platform.
+  // @todo Refactor for cleaner use of params.
+  await validateWebRoot(installOptions);
```

Comment:
_Nice._ I really like the way the chunks of this function were broken down, and the names are spot on. :+1: